File: redome-front/src/app/paciente/editar/contato.paciente.dto.ts
import {ContatoTelefonicoPaciente} from '../cadastro/contato/telefone/contato.telefonico.paciente';
import {EnderecoContatoPaciente} from '../cadastro/contato/endereco/endereco.contato.paciente';
/**
 * Classe que representa o contato do paciente, somente com as informações necessárias
 * para edição.
 * 
 * @author Pizão
 */
export class ContatoPacienteDTO {

	private _rmr:number;
	private _enderecosContato: EnderecoContatoPaciente[];
	private _contatosTelefonicos: ContatoTelefonicoPaciente[];
	private _email: string;


	public get enderecosContato(): EnderecoContatoPaciente[] {
		return this._enderecosContato;
	}

	public set enderecosContato(value: EnderecoContatoPaciente[]) {
		this._enderecosContato = value;
	}

	public get contatosTelefonicos(): ContatoTelefonicoPaciente[] {
		return this._contatosTelefonicos;
	}

	public set contatosTelefonicos(value: ContatoTelefonicoPaciente[]) {
		this._contatosTelefonicos = value;
	}

	public get email(): string {
		return this._email;
	}

	public set email(value: string) {
		this._email = value;
	}

	public get rmr(): number {
		return this._rmr;
	}

	public set rmr(value: number) {
		this._rmr = value;
	}
}

File: redome/src/main/java/br/org/cancer/modred/controller/dto/ContatoPacienteDTO.java
package br.org.cancer.modred.controller.dto;

import java.util.List;

import br.org.cancer.modred.model.ContatoTelefonicoPaciente;
import br.org.cancer.modred.model.EnderecoContatoPaciente;
import br.org.cancer.modred.model.Paciente;

/**
 * Classe representante do DTO de contato do paciente.
 * Esta classe traz somente o que é possível editar.
 * 
 * @author Pizão.
 *
 */
public class ContatoPacienteDTO {

	private Long rmr;
	private List<EnderecoContatoPaciente> enderecosContato;
	private List<ContatoTelefonicoPaciente> contatosTelefonicos;
	private String email;
	
	public ContatoPacienteDTO(){}
	
	/**
	 * Construtor sobrecarregado para que seja possível montá-lo a partir dos parâmetros informados.
	 * 
	 * @param rmr do paciente a ser associado ao dto.
	 * @param email do paciente a ser associado ao dto.
	 * @param enderecos endereços de contato do paciente a ser associado ao dto.
	 * @param telefones telefones de contato do paciente a ser associado ao dto.
	 */
	public ContatoPacienteDTO(Long rmr, String email, List<EnderecoContatoPaciente> enderecos, List<ContatoTelefonicoPaciente> telefones){
		this.rmr = rmr;
		this.email = email;
		this.enderecosContato = enderecos;
		this.contatosTelefonicos = telefones;
	}
	
	/**
	 * Construtor sobrecarregado para que seja possível montá-lo a partir dos parâmetros informados.
	 * 
	 * @param paciente a ser utilizado como base para criação do dto.
	 */
	public ContatoPacienteDTO(Paciente paciente){
		this(paciente.getRmr(), paciente.getEmail(), paciente.getEnderecosContato(), paciente.getContatosTelefonicos());
	}
	
	public Long getRmr() {
		return rmr;
	}
	public void setRmr(Long rmr) {
		this.rmr = rmr;
	}
	public List<EnderecoContatoPaciente> getEnderecosContato() {
		return enderecosContato;
	}
	public void setEnderecosContato(List<EnderecoContatoPaciente> enderecosContato) {
		this.enderecosContato = enderecosContato;
	}
	public List<ContatoTelefonicoPaciente> getContatosTelefonicos() {
		return contatosTelefonicos;
	}
	public void setContatosTelefonicos(List<ContatoTelefonicoPaciente> contatosTelefonicos) {
		this.contatosTelefonicos = contatosTelefonicos;
	}
	public String getEmail() {
		return email;
	}
	public void setEmail(String email) {
		this.email = email;
	}

	

}
